packages/components/src/input-control/input-field.tsx

```diff
@@ -82,16 +94,18 @@ function InputField(
 			return;
 		}
 		if ( ! isFocused && ! wasDirtyOnBlur.current ) {
-			update( valueProp );
+			update( valueProp, _event as SyntheticEvent );
 		} else if ( ! isDirty ) {
-			onChange( value, { event: _event } );
+			onChange( value, {
+				event: _event as ChangeEvent< HTMLInputElement >,
+			} );
```

Comment:
A minor thing but for quite some time I've had the opinion this should pass `_event` directly as the second argument. It would bring consistency with `onValidate`. Seems it wouldn't hurt to throw it in on this PR if you agree. On a quick search I didn't find any instances in Gutenberg this would even affect.

Reply:
It looks like the `UnitControl` actually does get used this way: https://github.com/WordPress/gutenberg/blob/8ef0f87acffd61f3e6357e3952192f1f2410c6b7/packages/components/src/box-control/input-controls.js#L40

It's tricky to find all the usages of `InputControl` as it's an "under the hood" component for many others.

It doesn't feel safe to make this change unfortunately.